fiftyone/server/routes/embeddings.py

```diff
@@ -235,42 +235,56 @@ async def post(self, request: Request, data: dict) -> dict:
         return await run_sync_task(self._post_sync, data)
 
     def _post_sync(self, data):
-        point_field = data["pointField"]
         dataset_name = data["datasetName"]
         stages = data["view"]
         patches_field = data["patchesField"]  # patches field of plot, or None
         selected_ids = data["selection"]  # selected IDs in plot
         slices = data["slices"]
-        lassoPoints = data.get("lassoPoints", None)
+        lasso_points = data.get("lassoPoints", None)
+        points_field = data.get("pointsField", None)
+
+        view = fosv.get_view(
+            dataset_name,
+            stages=stages,
+            sample_filter=get_sample_filter(slices),
+        )
+
+        is_patches_view = view._is_patches
+        is_patches_plot = patches_field is not None
+
+        if (
+            lasso_points is not None
+            and points_field is not None
+            and (patches_field is None or is_patches_view)
+        ):
+            # Lasso points via spatial index
+            # Unfortunately we can't use $geoWithin to filter nested arrays.
+            # The user must have switched to a patches view for this
+            if patches_field is not None:
+                _, points_field = view._get_label_field_path(
+                    patches_field, points_field
+                )
 
-        if point_field:
-            lasso_points_as_tuples = _get_fiftyone_geowithin(lassoPoints)
             stage = fos.Mongo(
                 [
                     {
                         "$match": {
-                            point_field: {
+                            points_field: {
```

Comment:
I used an `fos.Mongo` stage here just to get things working, however should we convert this to a `GeoWithin`?